crates/block-producer/src/psc.rs

```diff
@@ -271,6 +289,18 @@ async fn run(state: &mut ProduceSubmitConfirm) -> Result<()> {
             let context = state.context.clone();
             confirm_handle.replace_with(tokio::spawn(async move {
                 loop {
+                    match confirm_pending_l1_upgrade(&context).await {
+                        Ok(()) => (),
+                        Err(err) => {
+                            if err.is::<ShouldResyncError>() || err.is::<ShouldRevertError>() {
```

Comment:
If the upgrade tx cannot be confirmed, resync or revert may not be helpful.

Reply:
IMO throw the err is enough, What are you suggest to do here?